src/tests/ftest/tags.py

```diff
@@ -650,90 +747,210 @@ def print_verbose(*args):
-    parser.add_argument(
-        "command",
-        choices=("lint", "list", "dump", "unit"),
-        help="command to run")
-    parser.add_argument(
+    """Main function execution.
+
+    Returns:
+        int: 0 = success
+             1 = error
+    """
+    parser = ArgumentParser(prog='tags')
+    subparsers = parser.add_subparsers(
+        title='options for the tags command',
+        dest='command')
+
+    # Parser for the "config" command and its optional arguments
+    config_parser = subparsers.add_parser(
+        'config',
+        help='list test files and their requested test configuration values')
+    config_parser.add_argument(
+        "--paths",
+        nargs="+",
+        default=[],
+        help="file paths")
+    config_parser.add_argument(
+        "--tags",
+        nargs="+",
+        type=__comma_separated_arg,
+        help="tags used to filter which files to display")
+    config_parser.add_argument(
+        "--keys",
+        nargs="+",
+        type=__comma_separated_arg,
+        help="keys to display from the test yaml")
+    config_parser.add_argument(
+        "--csv",
+        type=str,
+        default=None,
+        help="csv output file")
+
+    # Parser for the "dump" command and its optional arguments
+    dump_parser = subparsers.add_parser(
+        'dump',
+        help='dump the file/class/method/tag structure for test files')
+    dump_parser.add_argument(
+        "--paths",
+        nargs="+",
+        default=[],
+        help="file paths")
+    dump_parser.add_argument(
+        "--tags",
+        nargs="+",
+        type=__comma_separated_arg,
+        help="tags used to filter which files to display")
+
+    # Parser for the "lint" command and its optional arguments
+    lint_parser = subparsers.add_parser(
+        'lint',
+        help='lint ftest avocado tags')
+    lint_parser.add_argument(
+        "--paths",
+        nargs="+",
+        default=[],
+        help="file paths")
+    lint_parser.add_argument(
         "-v", "--verbose",
         action='store_true',
         help="print verbose output for some commands")
-    parser.add_argument(
+
+    # Parser for the "list" command and its optional arguments
+    list_parser = subparsers.add_parser(
+        'list',
+        help='list ftest avocado tags associated with test files')
+    list_parser.add_argument(
         "--paths",
         nargs="+",
         default=[],
         help="file paths")
-    parser.add_argument(
-        "--tags",
-        nargs="+",
-        type=__arg_type_tags,
-        help="tags")
+
+    # Parser for the "unit" command and its optional arguments
+    unit_parser = subparsers.add_parser(
+        'unit',
+        help='run self unit tests')
+    unit_parser.add_argument(
+        "-v", "--verbose",
+        action='store_true',
+        help="print verbose output")
+
     args = parser.parse_args()
     args.paths = list(map(os.path.realpath, args.paths))
```

Comment:
This doesn't work now with the subparsers that don't have paths. E.g.
```
./src/tests/ftest/tags.py unit
Traceback (most recent call last):
  File "/home/dbohning/daos-stack/daos/./src/tests/ftest/tags.py", line 959, in <module>
    sys.exit(main())
  File "/home/dbohning/daos-stack/daos/./src/tests/ftest/tags.py", line 933, in main
    args.paths = list(map(os.path.realpath, args.paths))
AttributeError: 'Namespace' object has no attribute 'paths'
```

Reply:
Ah, right. I meant to make that conditional.